client/verta/verta/_cli/registry/create.py

```diff
@@ -43,34 +43,44 @@ def create_model(model_name, label, visibility, workspace):
 @click.option("--model", help="Path to the model.")
 @click.option("--artifact", type=(str, str), multiple=True, help="Path to an artifact required for the model. The format is --artifact artifact_key path_to_artifact.")
 @click.option("--workspace", "-w", help="Workspace to use.")
-@click.option("--from-run", help="ID of the Experiment Run to enter into the model registry. This option cannot be provided alongside --model nor --artifact.")
+@click.option("--from-run", type=str, help="ID of the Experiment Run to enter into the model registry. This option cannot be provided alongside other options, except for --workspace.")
 def create_model_version(model_name, version_name, label, model, artifact, workspace, from_run):
     """Create a new registeredmodelversion entry.
     """
     if artifact and len(artifact) > len(set(map(lambda pair: pair[0], artifact))):
         raise click.BadParameter("cannot have duplicate artifact keys")
 
+    if from_run and (label or model or artifact):
+        raise click.BadParameter("--from_run cannot be provided alongside other options, except for --workspace")
+
     client = Client()
 
     try:
         registered_model = client.get_registered_model(name=model_name, workspace=workspace)
     except ValueError:
         raise click.BadParameter("model {} not found".format(model_name))
 
-    model_version = registered_model.get_or_create_version(name=version_name, labels=list(label))
+    if from_run:
+        model_version = registered_model.create_version_from_run(run_id=from_run, name=version_name)
+
+        if label:
+            for l in label:
+                model_version.add_label(l)
+    else:
```

Comment:
Instead of nesting everything into `else`, let's actually put a `return` at the end of this initial `if` to end the function early when `from_run`. Does that make sense?

Reply:
I think so. Fixed.